### ISSUE TITLE
Enhancement: Log-Transform Prices for Linear Regression Model

### ISSUE THREAD

**Issue by ankit02327** [ankit02327/stock-price#88]
Stocks compound, meaning their prices grow exponentially rather than linearly over time. Linear regression fundamentally assumes a linear relationship. By passing raw stock prices into our `linear_regression` algorithm, we severely handicap the model's accuracy, especially over long prediction horizons.

By simply applying `np.log()` to the target prices before fitting the model, we transform the exponential price curve into a linear one, which the model can fit perfectly. We then apply `np.exp()` to the model's output to convert the predictions back to normal prices. This mathematically aligns the linear model with how stocks actually behave, unlocking a massive boost in accuracy.

**Comment by ankit02327** [comment]
1. In `backend/algorithms/optimised/linear_regression/linear_regression.py`, locate the `fit()` and `partial_fit()` methods.
2. Apply `np.log(y)` to the target vector before passing it to `self.model.fit()` or `self.model.partial_fit()`.
3. In the `predict()` method, apply `np.exp()` to the model's output before returning the predictions.
4. Update the training metrics calculation in `fit()` to use the exponentiated predictions against the original `y` values.

**Comment by ankit02327** [comment]
We will mathematically align the model with stock compounding.

```python
    def fit(self, X: np.ndarray, y: np.ndarray) -> 'ModelInterface':
        # ... initialization ...
        X_scaled = self.scaler.fit_transform(X)
        
        # Log-transform target prices for linear growth
        y_log = np.log(y)
        self.model.fit(X_scaled, y_log)
        
        # Calculate metrics using inverse transformed predictions
        y_pred = np.exp(self.model.predict(X_scaled))
        mse = mean_squared_error(y, y_pred)
        # ...
        
    def predict(self, X: np.ndarray) -> np.ndarray:
        # ... validation ...
        X_scaled = self.scaler.transform(X)
        
        # Predict and inverse log-transform back to absolute prices
        predictions_log = self.model.predict(X_scaled)
        return np.exp(predictions_log)
